File: src/api/dotnet/FiniteDomainSort.cs
﻿/*++
Copyright (c) 2012 Microsoft Corporation

Module Name:

    FiniteDomainSort.cs

Abstract:

    Z3 Managed API: Finite Domain Sorts

Author:

    Christoph Wintersteiger (cwinter) 2012-11-23

Notes:
    
--*/

using System.Diagnostics;
using System;

namespace Microsoft.Z3
{
    /// <summary>
    /// Finite domain sorts.
    /// </summary>
    public class FiniteDomainSort : Sort
    {
        /// <summary>
        /// The size of the finite domain sort.
        /// </summary>
        public ulong Size
        {
            get
            {
                ulong res = 0;
                Native.Z3_get_finite_domain_sort_size(Context.nCtx, NativeObject, ref res);
                return res;
            }
        }

        #region Internal
        internal FiniteDomainSort(Context ctx, IntPtr obj)
            : base(ctx, obj)
        {
            Debug.Assert(ctx != null);
        }
        internal FiniteDomainSort(Context ctx, Symbol name, ulong size)
            : base(ctx, Native.Z3_mk_finite_domain_sort(ctx.nCtx, name.NativeObject, size))
        {
            Debug.Assert(ctx != null);
            Debug.Assert(name != null);

        }
        #endregion
    }
}

File: src/api/java/FiniteDomainSort.java
/**
Copyright (c) 2012-2014 Microsoft Corporation
   
Module Name:

    FiniteDomainSort.java

Abstract:

Author:

    @author Christoph Wintersteiger (cwinter) 2012-03-15

Notes:
    
**/

package com.microsoft.z3;

/**
 * Finite domain sorts.
 **/
public class FiniteDomainSort extends Sort
{
    /**
     * The size of the finite domain sort.
     * @throws Z3Exception on error
     **/
    public long getSize()
    {
        Native.LongPtr res = new Native.LongPtr();
        Native.getFiniteDomainSortSize(getContext().nCtx(), getNativeObject(), res);
        return res.value;
    }

    FiniteDomainSort(Context ctx, long obj)
    {
        super(ctx, obj);
    }

    FiniteDomainSort(Context ctx, Symbol name, long size)
    {
        super(ctx, Native.mkFiniteDomainSort(ctx.nCtx(), name.getNativeObject(),
                size));
    }
}
